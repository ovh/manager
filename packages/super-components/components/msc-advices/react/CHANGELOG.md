# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.1](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.1.0...@ovhcloud/msc-react-advices@1.1.1) (2023-12-05)

**Note:** Version bump only for package @ovhcloud/msc-react-advices





# [1.1.0](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.0.0...@ovhcloud/msc-react-advices@1.1.0) (2023-12-05)


### Features

* msc-advices ([670db53](https://github.com/ovh/manager/commit/670db53549b433db305fa4e6875d804dc2da56c3))
