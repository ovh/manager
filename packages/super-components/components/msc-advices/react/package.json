{
  "name": "@ovhcloud/msc-react-advices",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for msc-advices super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-advices#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-advices/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "npm run clean && npm run compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-advices": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-components": "^16.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
