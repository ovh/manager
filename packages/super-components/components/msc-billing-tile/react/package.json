{
  "name": "@ovhcloud/msc-react-billing-tile",
  "version": "1.1.1",
  "private": true,
  "description": "React specific wrapper for Billing Tile super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-billing-tile#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-billing-tile/react"
  },
  "license": "BSD-3-Clause",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-billing-tile": "^1.1.1",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
