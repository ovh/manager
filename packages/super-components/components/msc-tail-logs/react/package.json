{
  "name": "@ovhcloud/msc-react-tail-logs",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Tail Logs super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-tail-logs#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-tail-logs/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tail-logs": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
