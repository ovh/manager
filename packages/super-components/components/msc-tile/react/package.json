{
  "name": "@ovhcloud/msc-react-tile",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tile": "^1.0.0",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/react": "18.2.8",
    "@types/react-dom": "18.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^13.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
