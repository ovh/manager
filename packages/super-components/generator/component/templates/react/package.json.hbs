{
  "name": "@ovhcloud/msc-react-{{ packageName }}",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/{{ componentTag }}#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/{{ componentTag }}/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "prepare": "yarn msc:build",
    "msc:build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-{{ packageName }}": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "17.0.39",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^14.1.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
