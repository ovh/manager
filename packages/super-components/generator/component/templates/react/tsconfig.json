{
  "extends": "../../../tsconfig.react.json",
  "compilerOptions": {
    "outDir": "dist/esm",
    "declarationDir": "dist/types"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"]
}
