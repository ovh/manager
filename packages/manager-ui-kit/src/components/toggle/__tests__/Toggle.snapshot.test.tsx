import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import { Toggle, ToggleControl, ToggleLabel } from '..';

describe('Toggle Snapshot tests', () => {
  it('renders the component with default props and children', () => {
    const { container } = render(
      <Toggle>
        <ToggleControl />
        <ToggleLabel>Enable dark mode</ToggleLabel>
      </Toggle>,
    );
    expect(container).toMatchSnapshot();
  });
});
