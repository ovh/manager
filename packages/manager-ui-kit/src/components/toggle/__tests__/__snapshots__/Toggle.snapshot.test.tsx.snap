// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toggle Snapshot tests > renders the component with default props and children 1`] = `
<div>
  <label
    class="_toggle_2vo1m_2"
    data-ods="toggle"
    data-part="root"
    data-scope="switch"
    data-state="unchecked"
    dir="ltr"
    for="switch::r0::input"
    id="switch::r0:"
  >
    <span
      aria-hidden="true"
      class="_toggle-control_t6d2a_2"
      data-ods="toggle-control"
      data-part="control"
      data-scope="switch"
      data-state="unchecked"
      dir="ltr"
      id="switch::r0::control"
    >
      <span
        aria-hidden="true"
        class="_toggle-control__thumb_t6d2a_42"
        data-part="thumb"
        data-scope="switch"
        data-state="unchecked"
        dir="ltr"
        id="switch::r0::thumb"
      />
    </span>
    <input
      aria-labelledby=""
      id="switch::r0::input"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
      type="checkbox"
      value="on"
    />
    <span
      data-ods="toggle-label"
      data-part="label"
      data-scope="switch"
      data-state="unchecked"
      dir="ltr"
      id="switch::r0::label"
    >
      Enable dark mode
    </span>
  </label>
</div>
`;
